### ISSUE TITLE
0.5 issues from gitter

### ISSUE THREAD

**Issue by d3x0r** [amark/gun#299]
```
var Gun = require( "./" );

var a = Gun( { file: "a.json" } );
var b = Gun( { file : "b.json" } );
var c = Gun( { file : "c.json" } );
var ab = a.get( "key" );
var bb = b.get( "key" );
ab.map( (val,field)=>{console.log("a key got val:", field, val ) } );
bb.map( (val,field)=>{console.log("b key got val:", field, val ) } );
bb.put( { msg: "hello"} );

setTimeout( ()=>{
    var cb = c.get( "key" );
    cb.map( (val,field)=>{console.log("c key got val:", field, val ) } );
}, 1000 );
```
c.json is the only file with data. b is the only one with a put. 'c key got val' message logs after 1 second....
expected? I mean I can definatly avoid using the same root names in get("key") ... but I would expect filenames at least to stay with the gun instances?

---
```
var Gun = require( './' )

var gun = Gun( {file: "test3.json"} );
var siteDef = gun.get( "siteDef" );
var gunSite = gun.get( "siteDef:1" );

var o = {};
o[1] = { id:1,name:"Apple",address:"::1" };
siteDef.put( o ); 
gunSite.put( o[1] );
```
only the first put works. Reversing them makes it so the other works but always the first

**Comment by amark** [comment]
I think @rsaccon 's issue is also related to this, knocking out a fix for you/this (and I think @hillct found it too) is my highest code priority. Another sample:

```
const bob = gundb.get('foo/bob').put({ name: 'Bob' })
bob.val((value) => {
  console.log("bob")
  console.log(value)
})
const rob = gundb.get('foo/rob').put({ name: 'Rob' })
rob.val((value) => {
  console.log("rob")
  console.log(value)
})
const foo = gundb.get('foo/foo').put({ name: 'Foo' })
foo.val((value) => {
  console.log("foo")
  console.log(value)
})
```

**Comment by amark** [comment]
@d3x0r @hillct @rsaccon please check to see if this now works. The problem was with the file.js/data.json storage module.

**Comment by d3x0r** [comment]
no events fire now.

**Comment by d3x0r** [comment]
fixed since 0.6 I'm sure
